homeassistant/components/mqtt/fan.py

```diff
@@ -172,7 +174,25 @@ def valid_preset_mode_configuration(config):
     }
 ).extend(MQTT_ENTITY_COMMON_SCHEMA.schema)
 
+# The use of PLATFORM_SCHEMA is deprecated in HA Core 2022.6
 PLATFORM_SCHEMA = vol.All(
+    cv.PLATFORM_SCHEMA.extend(_PLATFORM_SCHEMA_BASE.schema),
+    valid_speed_range_configuration,
+    valid_preset_mode_configuration,
+    cv.deprecated(CONF_PLATFORM),  # Deprecated in HA Core 2022.6
```

Comment:
Here's an example from the template integration: <https://github.com/home-assistant/core/blob/7d26be2d4d0711796c0870d7917d68751aad7a66/homeassistant/components/template/config.py#L92-L98>

Reply:
> I don't think we should use the "deprecated" helper, because the resulting warning message is not helpful. We should use a custom validator instead, which warns once per MQTT platform that the MQTT config should be moved from the platform key (fan) to the integration key (mqtt).

Have already worked this out.